### PR TITLE
Add support for the Burnhard BBQ thermometer

### PR DESCRIPTION
**Pull request by fetzerch**
This is the implementation for the thermometer described in https://github.com/merbanan/rtl_433_tests/pull/383.

### CODE REVIEW

File: src/devices/burnhardbbq.c [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License version 3 as
+    published by the Free Software Foundation.
+*/
+/**
+Burnhard BBQ thermometer.
+
+Data format:
+
+1f 22 00 9052 44 1425e5 1e 8
```
Comment (zuckschwerdt) [inline src/devices/burnhardbbq.c:15]: indent these two lines 4 spaces

File: src/devices/burnhardbbq.c [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+AA SD ?? TTTT mt XXYXYY CC ?
+
+AA   device code (changes when battery is removed)
```
Comment (zuckschwerdt) [inline src/devices/burnhardbbq.c:18]: prefix each item with a dash+space

File: src/devices/burnhardbbq.c [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+YYY  temperature (-500, /10)
+CC   CRC
+?    a single bit (coding artefact)
+*/
+
+#include "decoder.h"
+
+static int burnhardbbq_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    uint8_t b[10];
```
Comment (zuckschwerdt) [inline src/devices/burnhardbbq.c:35]: make this `   uint8_t *b;`

File: src/devices/burnhardbbq.c [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    if (decoder->verbose > 1) {
+        bitbuffer_printf(bitbuffer, "%s: ", __func__);
+    }
+
+    // All three rows contain the same information. Return on first decoded row.
+    for (int i = 0; i < bitbuffer->num_rows; ++i) {
+        // A row typically has 81 bits, but the last is just a coding artefact.
+        if (bitbuffer->bits_per_row[i] < 80)
+            continue;
+        bitbuffer_extract_bytes(bitbuffer, i, 0, b, 80);
```
Comment (zuckschwerdt) [inline src/devices/burnhardbbq.c:54]: make this `b = bitbuffer->bb[i];`

File: src/devices/burnhardbbq.c [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        // Sanity check (digest last byte).
+        if (lfsr_digest8_reflect(b, 9, 0x31, 0xf4) != b[9])
+            continue;
+
+        id = b[0];
+        channel = b[1] & 0x07;
+        temperature_alarm = (b[1] & 0x80) > 0;
+        timer_alarm = (b[1] & 0x40) > 0;
+        timer_active = (b[1] & 0x10) > 0;
+        temperature_setpoint = ((b[6] | (b[7] & 0x0f) << 8) - 500) / 10.0f;
```
Comment (zuckschwerdt) [inline src/devices/burnhardbbq.c:65]: `* 0.1f` is preferred over `/ 10.0f`

**Comment by fetzerch** [comment]
Thx, fixed.

**Comment by Mindavi** [comment]
@fetzerch was it intended to license this as gplv3only? Would you be willing to relicense under GPLv2+?

Ping @zuckschwerdt including GPLv3 only code will make the build gplv3, unless this source file is not compiled in.

**Comment by zuckschwerdt** [comment]
Oops. That snuck by me. GPL2+ is preferred. @fetzerch can you PR the license change, if you deem that ok?

**Comment by fetzerch** [comment]
Sure: https://github.com/merbanan/rtl_433/pull/1642. Please note that this was completely unintentional. I always prefer to stick to a project's main license.

**Comment by Mindavi** [comment]
It happens. Thanks for the PR :)

**Pull request merged**